server/functions.lua

```diff
@@ -83,3 +83,140 @@ function ToggleDoorState(entity)
+
+    return player.PlayerData.citizenid
+end
+
+---Looking for a vehicle in the world
+---@param plate string The plate number of the vehicle.
+---@return number? vehicle The entity number of the found vehicle, nil otherwise.
+function public.findVehicleByPlate(plate)
+    local vehicles = GetAllVehicles()
+    for i = 1, #vehicles do
+        local vehicle = vehicles[i]
+        if qbx.getVehiclePlate(vehicle) == plate then
+            return vehicle
+        end
+    end
+end
+
+---Loads a players vehicles to the vehicleList
+---@param src integer
+function public.addPlayer(src)
+    local citizenid = getCitizenId(src)
+    if not citizenid then return end
+
+    local vehicles = MySQL.query.await('SELECT * FROM player_vehicles WHERE citizenid = ?', { citizenid })
+
+    local state = {}
+    local platesAssociations = {}
+
+    for i = 1, #vehicles do
+        platesAssociations[vehicles[i].plate] = true
+    end
+
+    if keysList[citizenid] then
+        keysLifetime[citizenid] = nil
+
+        for plate in pairs(keysList[citizenid]) do
+            platesAssociations[plate] = true
+        end
+    end
+
+    local worldVehicles = GetAllVehicles()
+    for i = 1, #worldVehicles do
+        local vehiclePlate = qbx.getVehiclePlate(worldVehicles[i])
+        if platesAssociations[vehiclePlate] then
+            state[vehiclePlate] = true
+        end
+    end
+
+    Player(src).state:set('keysList', state, true)
+end
+
+---Removes a players vehicles from the vehicleList
+---@param src integer
+function public.removePlayer(src)
+    local citizenid = getCitizenId(src)
+    if not citizenid then return end
+
+    keysList[citizenid] = Player(src).state['keysList']
+    keysLifetime[citizenid] = os.time()
+
+    Player(src).state:set('keysList', nil, true)
+end
+
+--- Removing the vehicle keys from the user
+---@param source number ID of the player
+---@param plate string The plate number of the vehicle.
+function public.removeKeys(source, plate)
+    local citizenid = getCitizenId(source)
+
+    if not citizenid then return end
+
+    local keys = Player(source).state.keysList or {}
+
+    if not keys[plate] then return end
+    keys[plate] = nil
+
+    Player(source).state:set('keysList', keys, true)
+
+    exports.qbx_core:Notify(source, locale('notify.keys_removed'))
+
+    return true
+end
+
+function public.hasKeys(source, plate)
+    return Player(source).state.keysList[plate]
+end
+
+---Gives the user the keys to the vehicle
+---@param source number ID of the player
+---@param plate string The plate number of the vehicle.
+function public.giveKeys(source, plate)
+    local citizenid = getCitizenId(source)
+
+    if not citizenid then return end
+
+    local keys = Player(source).state.keysList or {}
+
+    if keys[plate] then return end
+    keys[plate] = true
+
+    Player(source).state:set('keysList', keys, true)
+
+    exports.qbx_core:Notify(source, locale('notify.keys_taken'))
+
+    return true
+end
```

Comment:
These should also update the keys lists

Reply:
These shouldn't

Reply:
These shouldn't.

Reply:
Why not?

Reply:
this holds the state of the keys only log off characters